use gtfsio backend and improve hms conversion

polettif opened the pull request:
With this PR [gtfsio](https://github.com/r-transit/gtfsio) is used for reading/writing gtfs feeds. `gtfsio` provides fast functions based on data.table. This PR is currently a work in progress, depending on possible changes from `gtfsio` version 0.1.0.

Deliberate design decisions and differences to `gtfsio`:
- tidytransit gtfs objects are of class `tidygtfs`, inheriting from gtfsio's `gtfs` (breaking change, currently tidytransit uses gtfs as well)
- instead of `data.tables`, tidytransit uses `tibbles` for tables of a feed
- dates (e.g. in calendar.txt) are converted to `Dates`
- times (e.g. in stop_times.txt) are converted to `hms` times by default. They replacing the existing character columns (breaking change to current version). Basically `g <- read_gtfs("feed.zip") %>% set_hms_times() %>% set_date_service_table()` is replaced by `g <- read_gtfs("feed.zip")`
- spatial data (stops and shapes) are not converted upon reading, this can still be done with `gtfs_as_sf`

Thanks to some performance improvements with calculating hms times and gtfsio's fast reading implementation it's now reasonable to convert times/dates upon reading by default. Converting times/dates previously took up to 5 times as long as it does now.

Also tidytransit now creates the table `dates_services` automatically (previously called date_service_table). It is still stored in the `.` sublist: 
```r
g$.$dates_services
# A tibble: 3,349 x 2
   date       service_id             
   <date>     <chr>                  
 1 2018-06-24 ASP18GEN-1037-Sunday-00
 2 2018-06-24 ASP18GEN-2048-Sunday-00
 3 2018-06-24 ASP18GEN-3041-Sunday-00
# … with 3,346 more rows
```

Other auxiliary tables are renamed as well: service_pattern to `servicepatterns` and date_servicepattern_table to `dates_servicepatterns`. They are created with `set_servicepattern` (unchanged).

polettif committed:
use gtfsio::import_gtfs

polettif committed:
remove validation test

polettif committed:
sample-feed-plus .csv as .txt

polettif committed:
extend start/end date checks

polettif committed:
add validation

polettif committed:
add tidygtfs class

inherits from gtfs

polettif committed:
faster hh:mm:ss conversion

33s -> 4s

polettif committed:
use new_hms

polettif committed:
use dt in set_dates

polettif committed:
set_hms_times used by default

polettif committed:
overwrite arrival_time and departure_time w/ hms

fix wrongly formatted hhmmss strings

polettif committed:
fix vignettes

polettif committed:
handle possibly malformed time strings

polettif committed:
update vignettes for new behavior

polettif committed:
rename date_servcie_table to dates_services

polettif committed:
rename date_servicepattern_table to dates_servicepatterns

polettif committed:
rename service_pattern to servicepatterns

polettif committed:
fix filenames in validate_gtfs

polettif committed:
print.tidygtfs

polettif committed:
Merge branch 'dev/aux-tables' into dev/gtfsio

polettif committed:
remove unused transitfeeds functions

polettif committed:
update write_gtfs

tbuckl commented:
this is fast 💨 . leaving a comment in 1 spot.

tbuckl commented on tests/testthat/test-headways.R:
```diff
@@ -1,25 +1,20 @@
-      stops_frequency$stop_id==778123, ]$headway
-  expect_equal(as.integer(14), as.integer(ex_address))
+    stops_frequency$headway[stops_frequency$stop_id==778037]
+  expect_equal(as.integer(240), as.integer(ex_address))
```
@polettif i think is a bad test that may be my fault. i can remove if helpful. part of whats confusing about the test, and perhaps even the syntax of the function, is that it refers to seems to refer to a single "stop_id" but that can refer to multiple services and, i think, multiple directions. e.g. see how at the end of the vignette for NYC i reduced the final graphic down to stop names.

polettif replied:
Thanks @tbuckl, I wasn't quite sure how to handle the test. Another problem is that the dataset has been updated so it doesn't really work as a regression test anymore. Do you know a feed/setup where we know the "actual" stop frequency?

Removing it would be an easy way to handle it but I think we should cover `get_route_frequency` with an adequate test, especially since the syntax might be confusing.

polettif replied:
Another note on frequencies (and maybe a tangent): I'm not really happy with the `dow` (day of the week) parameter since it assumes that a feed only uses calendar.txt in `service_by_dow`.

polettif committed:
use tempfile

remove gtfstools duplicates

polettif committed:
remove references to set_date_services

polettif committed:
update gtfs_duke dataset to new format

polettif committed:
gtfs_as_sf with crs

polettif committed:
update get_stop_frequency

remove get_route_frequency functions
add line breaks
replace functions in vignette

polettif committed:
suggest markdown for knitr::kable()

https://github.com/yihui/knitr/issues/1864

polettif committed:
export write_gtfs

remove is_gtfs_obj

polettif committed:
update docs

internal set_dates_services

polettif commented:
@tbuckl I updated `get_stop_frequency`, the frequency vignette and the corresponding test. I also removed `get_route_frequency` as I'm not sure it really did what it was supposed to. There's still route frequency calculations in the vignette but it's no longer contained in a separate function. Is that ok for you?

Other significant changes besides bugfixes and updated docs:
- `write_gtfs` is exported, closes #163 
- `gtfs_as_sf` has a crs (coordinate reference system) parameter to transform coordinates right away

IMO this PR is good to merge. I think a version bump to 1.0 is reasonable, considering all the breaking changes and new gtfsio implementation.

tbuckl commented:
@polettif yes i'm good with whatever in terms of the vignette--maybe lets just remove the whole thing for now we can always add it back in later.

tbuckl committed:
remove frequency vignette

until we have another ready

tbuckl committed:
bump version

tbuckl commented:
merged and submitted to CRAN.

polettif commented:
👍 thank you
